cpp/src/arrow/util/tracing_internal.h

```diff
@@ -114,23 +116,44 @@ class SpanImpl {
 opentelemetry::trace::StartSpanOptions SpanOptionsWithParent(
     const util::tracing::Span& parent_span);
 
-#define START_SPAN(target_span, ...)                                                \
-  auto opentelemetry_scope##__LINE__ =                                              \
-      ::arrow::internal::tracing::GetTracer()->WithActiveSpan(                      \
-          target_span                                                               \
-              .Set(::arrow::util::tracing::Span::Impl{                              \
-                  ::arrow::internal::tracing::GetTracer()->StartSpan(__VA_ARGS__)}) \
-              .span)
-
-#define START_SPAN_WITH_PARENT(target_span, parent_span, ...)                           \
-  auto opentelemetry_scope##__LINE__ =                                                  \
-      ::arrow::internal::tracing::GetTracer()->WithActiveSpan(                          \
-          target_span                                                                   \
-              .Set(::arrow::util::tracing::Span::Impl{                                  \
-                  ::arrow::internal::tracing::GetTracer()->StartSpan(                   \
-                      __VA_ARGS__,                                                      \
-                      ::arrow::internal::tracing::SpanOptionsWithParent(parent_span))}) \
-              .span)
+#define CONFIGURE_TARGET_SPAN(target_span, ...)                             \
+  target_span                                                               \
+      .Set(::arrow::util::tracing::Span::Impl{                              \
+          ::arrow::internal::tracing::GetTracer()->StartSpan(__VA_ARGS__)}) \
+      .span
+
+#define START_COMPUTE_SPAN(target_span, ...)                                      \
+  auto modified_span = CONFIGURE_TARGET_SPAN(target_span, __VA_ARGS__);           \
+  modified_span->SetAttribute("memory_pool_bytes",                                \
```

Comment:
nit, but should we perhaps namespace this attribute name? "arrow.memory_pool_bytes" or something

Reply:
Note how the official ones are semi-namespaced: https://github.com/open-telemetry/opentelemetry-cpp/blob/4f5505f9cb45326299067cfbd53db21198970aa8/sdk/include/opentelemetry/sdk/resource/experimental_semantic_conventions.h